### ISSUE TITLE
CRASH - something snarky - parent missing from frontier

### ISSUE THREAD

**Issue by jkrauska** [CodaProtocol/coda#3349]
Error

```
{
  "timestamp": "2019-09-01 12:58:44.350454Z",
  "level": "Fatal",
  "source": {
    "module": "Init__Coda_run",
    "location": "File \"app/cli/src/init/coda_run.ml\", line 411, characters 56-63"
  },
  "message": "Unhandled top-level exception: $exn\nGenerating crash report",
  "metadata": {
    "exn": "\"Error validating proposed transition frontier dependencies: parent missing from frontier\"",
    "pid": 30556
  }
}
```

Log:
[test-coda-CRASH-2019-09-01_13-00-06_coda.log.gz](https://github.com/CodaProtocol/coda/files/3576057/test-coda-CRASH-2019-09-01_13-00-06_coda.log.gz)

**Comment by imeckler** [comment]
@enolan 

Cause:

We build a block on our current best tip, but in the mean time our best tip moved and the parent is not there.

Immediate fix:
Don't crash and just print an error if this happens. 

Long term fix:
If our best tip moves, kill the current proposer.

**Comment by imeckler** [comment]
@ghost-not-in-the-shell will take a look because https://github.com/CodaProtocol/coda/pull/3152 should have fixed this.

**Comment by imeckler** [comment]
This is also related to an RFC he has

**Comment by ghost-not-in-the-shell** [comment]
This crash should be fixed by #3152. I just checked that my PR was not included in this release, but it would be included in the future release.
